packages/patrol_cli/test/crossplatform/app_options_test.dart

```diff
@@ -103,6 +111,75 @@ void main() {
         );
       });
     });
+
+    group('correctly overrides dart defines from file values with dart defines',
+        () {
+      const dartDefines = {
+        'EMAIL': 'user@example.com',
+        'PASSWORD': 'ny4ncat',
+        'foo': 'bar',
+      };
+
+      const dartDefinesFromFile = {
+        'EMAIL': 'banana@example.com',
+        'PASSWORD': 'hello',
+        'FLAVOR': 'dev'
+      };
+
+      test('on Windows', () {
+        final flutterOpts = FlutterAppOptions(
+          target: r'C:\Users\john\app\integration_test\app_test.dart',
+          buildMode: BuildMode.debug,
+          flavor: 'dev',
+          dartDefines: dartDefines,
+          dartDefineFromFile: dartDefinesFromFile,
+          dartDefineFromFilePath: '',
+        );
+        options = AndroidAppOptions(flutter: flutterOpts);
+
+        final invocation =
+            options.toGradleAssembleTestInvocation(isWindows: true);
+        expect(
+          invocation,
+          equals([
+            r'.\gradlew.bat',
+            ':app:assembleDevDebugAndroidTest',
+            r'-Ptarget=C:\Users\john\app\integration_test\app_test.dart',
+            '-Pdart-defines=RU1BSUw9dXNlckBleGFtcGxlLmNvbQ==,UEFTU1dPUkQ9bnk0bmNhdA==,Zm9vPWJhcg==',
+            '-PEMAIL=user@example.com',
+            '-PPASSWORD=ny4ncat',
+            '-PFLAVOR=dev',
```

Comment:
That's a nice test (and below as well!), but I think it's slightly wrong.

First of all, it's not needed to pass these values twice: once in `Pdart-defines` (in base64-encoded form) and once again in `Pkey=value`. Specifying them in `-Pdart-defines` is enough.

And second possible problem: there are only 3 defines in `-Pdart-defines`: `EMAIL`, `PASSWORD`, and `foo`. But there's also `FLAVOR`, and I don't see it encoded in `-Pdart-defines`. I guess this probably stems from the first problem.

Reply:
oooh okay, got you! I'm worried tho about not passing both (I don't know if you remember my issue) but since my app was launched using `--dart-defines-from-file` it wouldn't take the `--dart-defines` values as valid. So in that case, wouldn't it be better to send them in the `PKEY=VALUE` format without the `Pdart-defines=KEY=VALUE`? hope that makes sense.

In the case of FLAVOR it was that it was in the config file but not in the dart-defines, so it shouldn't be taking it anyways? So it overrides all the values from the dart defines, but it also keeps the ones that aren't passed

Reply:
nvm, I realized that I usually use either the config file or the dart defines, not sure if the first point still applies.
The second one, I'm still curious about